javascript/src/async/zones.js

```diff
@@ -27,6 +38,25 @@ const zoneGeometrySelect = ({ runConfiguration }) => {
 }
 
 export default store => {
+    // Available Zones
+    resync(store, availableZoneSelect, ({ point, method }, io, dispatch) => {
+        let pointIsValid = point !== null && point.x && point.y
```

Comment:
Very unlikely, but `point.x = 0` would result in this check failing when it shouldn't. Better to do explicit comparison: ```point.x !== null```.

Reply:
Also stolen code.. I'll remove from both.

Reply:
Well, not a remove so much as an add :)